iib/web/static/api_v1.yaml

```diff
@@ -210,30 +210,77 @@ paths:
               $ref: '#/components/schemas/RequestUpdate'
       security:
         - Kerberos Authentication: []
-  /healthcheck:
+  /healthcheck/details:
     get:
-      description: Perform a health check of the service
+      description: Get the status of the services IIB depends on
       responses:
         '200':
-          description: The health check OK message
+          description: services status information
           content:
             application/json:
               schema:
                 type: object
                 properties:
-                  status:
-                    type: string
-                    example: Health check OK
-        '500':
-          description: Health check failed
+                  services:
+                    type: array
+                    items:
+                      type: object
+                      properties:
+                        name:
+                          type: string
+                          description: Name of a service
+                          example: database
+                        available:
+                          type: boolean
+                          description: indicator that service is running fine.
+                          example: false
+                        reason:
+                          type: string
+                          description: Reason why service is not running.
+                          example: Database connection failed
+                    required:
+                      - name
+                      - available
+                  workers:
+                    type: array
+                    items:
+                      type: object
+                      properties:
+                        name:
+                          type: string
+                          description: Name of a worker
+                          example: celery@025f84b0154f
+                        available:
+                          type: boolean
+                          description: indicator that worker is available.
+                          example: false
+                        reason:
+                          type: string
+                          description: Reason why worker is not available.
+                          example: worker didn't respond to ping
+                    required:
+                      - name
+                      - available
+            required:
+              - services
+              - workers
+  /healthcheck:
+    get:
+      summary: Get a short IIB service status
+      description: >
+        if services including RabbitMQ, database, and workers
+        are working properly, return 200.
+      responses:
+        '200':
+          description: services and workers are working as expected.
           content:
             application/json:
-              schema:
```

Comment:
Do we know if nobody uses the schema to automate if IIB works or not? 
I really like to use schema to at least show to have structured output.

Reply:
Can you elaborate more here how this would be used to automate if IIB works or not? 

Isn't ("Health Check OK"  or "Health Check failed") what would be used for validating the status of iib. am I missing something here?

Reply:
The response  `example: {'status': 'Health check OK'}` is ok but the schema says that it is a json object. Having just an example doesn't say if it is a string or object.

I mean Semaphore used the JSON from the healthcheck to show if IIB is working. I am not sure if there are any similar scripts

Reply:
My understanding is that this file is only for documentation. correct ? 

I know Semaphore would be shutdown soon, and I am not familiar how it checks if iib is working or not based on the schema. So, I am looping @midnightercz as he knows more and can tell if it would break something or not. wdyt Jindrich?

Reply:
@zanssa If we accept this change this new/modified endpoint will be the only one that does not have it documented.
And you are saying "only for documentation"? I mean - you never relied on documentation? Are you always looking to the code to see how to work with any API that you are using?

I agree with @pbortlov that it is good to have it documented. Why we should lower our standards with this PR?

Could you add the schema here, please?

Reply:
Oh! I think the change is already approved :)

There is a difference between using the documentation as instructions and using the schema there to test if iib is working on not.

What is your standard that is lowered with this PR?

Reply:
For me the standard for every documented endpoint is to have this schema - therefore I would assume this one should have it as well.

Reply:
@zanssa It shouldn't matter if the change is approved. One person might miss something which the other would notice. Please do address comments from all reviewers. We shouldn't merge a change unless concerns of all our guild members are resolved. 

fwiw, I agree we shouldn't cut corners in documentation and keep it as accurate as possible. One use case could be if another service uses IIB's endpoint to check its status and assumes it's going to be string instead of JSON from the documentation.

Reply:
@yashvardhannanavati I addressed the comments from all and I even took into consideration what I was asked in gchat instead of here, and despite that some of them are not really necessary or even enforced in the repo.

I would expect some change requests and I don't really mind making the changes but it would be nice if those change requests are mentioned after reviewing once or twice; not one change every few days. This is just slowing the progress and not helping! :)